src/app/components/register/register.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, inject } from '@angular/core';
+import {
+  NzFormDirective,
+  NzFormControlComponent,
+  NzFormItemComponent,
+} from 'ng-zorro-antd/form';
+import { NzInputDirective, NzInputGroupComponent } from 'ng-zorro-antd/input';
+import { NzColDirective, NzRowDirective } from 'ng-zorro-antd/grid';
+import { NzButtonComponent } from 'ng-zorro-antd/button';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  NonNullableFormBuilder,
+  ReactiveFormsModule,
+  ValidatorFn,
+  Validators,
+} from '@angular/forms';
+import { Router } from '@angular/router';
+import { DesktopGraphicsComponent } from '../desktop-graphics/desktop-graphics.component';
+
+@Component({
+  selector: 'app-register',
+  standalone: true,
+  templateUrl: './register.component.html',
+  styleUrl: './register.component.scss',
+  imports: [
+    NzFormDirective,
+    NzFormControlComponent,
+    NzFormItemComponent,
+    NzInputDirective,
+    NzInputGroupComponent,
+    NzColDirective,
+    NzRowDirective,
+    NzButtonComponent,
+    ReactiveFormsModule,
+    DesktopGraphicsComponent,
+  ],
+})
+export class RegisterComponent {
+  router = inject(Router);
+  validateForm: FormGroup<{
+    name: FormControl<string>;
+    email: FormControl<string>;
+    password: FormControl<string>;
+    checkPassword: FormControl<string>;
+  }>;
+
+  submitForm(): void {
+    if (this.validateForm.valid) {
+      // console.log('submit', this.validateForm.value);
+    } else {
+      Object.values(this.validateForm.controls).forEach((control) => {
+        if (control.invalid) {
+          control.markAsDirty();
+          control.updateValueAndValidity({ onlySelf: true });
+        }
+      });
+    }
+  }
+
+  updateConfirmValidator(): void {
+    /** wait for refresh value */
+    Promise.resolve().then(() =>
+      this.validateForm.controls.checkPassword.updateValueAndValidity()
+    );
+  }
+
+  confirmationValidator: ValidatorFn = (
+    control: AbstractControl
+  ): { [s: string]: boolean } => {
+    if (!control.value) {
+      return { required: true };
+    } else if (control.value !== this.validateForm.controls.password.value) {
+      return { confirm: true, error: true };
+    }
+    return {};
+  };
+
+  getCaptcha(e: MouseEvent): void {
+    e.preventDefault();
+  }
+
+  constructor(private fb: NonNullableFormBuilder) {
+    this.validateForm = this.fb.group({
+      name: ['', [Validators.required]],
+      email: ['', [Validators.email, Validators.required]],
+      password: ['', [Validators.required]],
```

Comment:
consider adding a pattern validator to the password for security purposes

Reply:
I will do so, these are still just from the UI library I used, I will update the validation as soon as possible.